packages/model-viewer/src/features/scene-graph.ts

```diff
@@ -100,20 +101,32 @@ export const SceneGraphMixin = <T extends Constructor<ModelViewerElementBase>>(
       return onUpdate
     }
 
+    // Load texture helper waits for the texture and image to finish loading
+    // before returning.
+    async[$loadTexture](uri: string) {
+      let result: Texture|null = null;
+      await new Promise(resolve => {
+        return new TextureLoader().load(uri, texture => {
+          resolve(texture);
+        });
+      }).then(completedTexture => {
```

Comment:
`await` is the modern version of `.then()`. There should be no need to use both, so we don't have `.then()` in our code base.

Reply:
Done, this worked
'''
await new Promise<Texture>(
          (resolve) => this[$textureLoader].load(uri, resolve));
'''
 I was hitting an issue where texture.img was undefined, not fully loaded